cpp/independent_set/IndependentSet.cpp

```diff
@@ -0,0 +1,318 @@
+
+#include <networkit/auxiliary/BucketPQ.hpp>
+#include <networkit/graph/GraphTools.hpp>
+
+#include <limits>
+#include <bitset>
+
+namespace Koala {
+
+void printGraph(const NetworKit::Graph* graph) { // TODO: its just debug
+    graph->forNodes([&](NetworKit::node v) {
+        std::cout << v << " ";
+    });
+    std::cout << std::endl;
+    graph->forEdges([&](NetworKit::node u, NetworKit::node v) {
+        std::cout << "{" << u << "," << v << "}, ";
+    });
+    std::cout << std::endl;
+}
+
+std::vector<NetworKit::node> IndependentSet::getNeighbors(NetworKit::node v) const {
+    std::vector<NetworKit::node> neighbors;
+    graph->forNeighborsOf(v, [&](NetworKit::node u) {
+        neighbors.push_back(u);
+    });
+    return neighbors;
+}
+
+std::vector<NetworKit::node> IndependentSet::getNeighborsPlus(NetworKit::node v) const {
+    std::vector<NetworKit::node> neighborsPlus = getNeighbors(v);
+    neighborsPlus.push_back(v);
+    return neighborsPlus;
+}
+
+std::vector<NetworKit::node> IndependentSet::getNeighbors2(NetworKit::node v) const {    
+    std::vector<NetworKit::node> neighbors = getNeighbors(v);
+    std::vector<NetworKit::node> visitedNodes;    
+    for (auto n : neighbors) {
+        graph->forNeighborsOf(n, [&](NetworKit::node n2) {
+            visitedNodes.push_back(n2);
+        });
+    }
+
+    std::vector<bool> visitedOnGraph(graph->numberOfNodes());
+    visitedOnGraph[v] = true;
+    for (auto x : neighbors) {
+        visitedOnGraph[x] = true;
+    }
+
+    std::vector<NetworKit::node> neighbors2;
+    for (auto x : visitedNodes) {
+        if (!visitedOnGraph[x]) {
+            visitedOnGraph[x] = true;
+            neighbors2.push_back(x);
+        }
+    }
+    return neighbors2;
+}
+
+IndependentSet::EdgeSet IndependentSet::getConnectedEdges(std::vector<NetworKit::node>& nodes) const {
+    EdgeSet connectedEdges(edgeComparator);
+    for (auto x : nodes) {
+        graph->forEdgesOf(x, [&](NetworKit::node u, NetworKit::node v) {
+            connectedEdges.insert(NetworKit::Edge(u, v, true));
+        });
+    }
+    return connectedEdges;
+}
+
+IndependentSet::EdgeSet IndependentSet::getInducedEdges(std::vector<NetworKit::node>& nodes) const {
+    EdgeSet connectedEdges(edgeComparator); 
+    std::set<NetworKit::node> nodeSet(nodes.begin(), nodes.end());
+
+    for (auto u : nodes) {
+        graph->forEdgesOf(u, [&](NetworKit::node u, NetworKit::node v) {
+            if (nodeSet.contains(v)) {
+                connectedEdges.insert(NetworKit::Edge(u, v, true));
+            }
+        });
+    }
+    return connectedEdges;
+}
+
+IndependentSet::EdgeSet IndependentSet::getAllEdges() const {
+    EdgeSet edges(edgeComparator);
+    graph->forEdges([&](NetworKit::node u, NetworKit::node v) {
+        edges.insert(NetworKit::Edge(u, v, true));
+    });
+    return edges;
+}
+
+std::vector<NetworKit::node> IndependentSet::getAllNodes() const {
+    std::vector<NetworKit::node> nodes;
```

Comment:
return std::vector<>(graph->nodeRange.begin(), graph->nodeRange.end())   should work all the same